File: example/testclient.cc
#include <iostream>
#include <cassert>
#include <cstring>

#include "ctflags.h"

/*
 * From the root directory, build the Rust library with:
 *    cargo build
 * then build the example with:
 *    g++ -o ./testclient example/testclient.cc -Iinclude -Ltarget/debug -lctflags -lpthread -ldl -lm
 */
int main() {
    ctflags::Seed seed;
    if (seed) {
        std::cout << "Seed is : " << seed.get() << std::endl;
    }

    ctflags::Flag flag("example");
    if (flag) {
        std::cout << "Example flag: " << flag.get() << std::endl;
    }

    // Test with a string context
    ctflags::Flag flag2("segg1545", "example", nullptr);
    std::cout << "Flag with string context is " << flag2.get() << std::endl;
    assert(strcmp(flag2.get(), "flag(example).5f1b958992ca66c09c0ac9170fce85de") == 0);

    return 0;
}

File: example/testclient.py
import ctflags

def main():
    # Test the seed
    seed = ctflags.get_seed_or_null()
    if seed:
        print(f"Seed is : {seed}")

    # Test the flag
    flag = ctflags.format_flag("example")
    print(f"Example flag: {flag}")

    # Salted flag
    flag_salt = ctflags.format_flag("example", "mon_sel")
    print(f"Flag with salt: {flag_salt}")

    # Test with a string context
    with_string_context = ctflags.format_flag_from_context("segg1545", "example")
    print(f"Flag with string context is {with_string_context}")
    assert with_string_context == "flag(example).5f1b958992ca66c09c0ac9170fce85de"

if __name__ == "__main__":
    main()
